drivers/flash/Kconfig.sam

```diff
@@ -9,8 +9,8 @@ config SOC_FLASH_SAM
 	select FLASH_HAS_PAGE_LAYOUT
 	select FLASH_HAS_DRIVER_ENABLED
 	select MPU_ALLOW_FLASH_WRITE if ARM_MPU
-	depends on SOC_FAMILY_SAM || \
-		   SOC_SERIES_SAME70 || \
+	depends on SOC_FAMILY_SAM
+	depends on SOC_SERIES_SAME70 || \
```

Comment:
We should clean this up to not have 2 `depends on`

Reply:
Verified and this should be:

`depends on SOC_SERIES_SAME70 || SOC_SERIES_SAMV71`

Reply:
done

Reply:
k, done

Reply:
Hi @galak ,

Ok, but samples/subsys/usb/dfu still selects SAM4L.

```
./scripts/twister -p sam4l_ek
Renaming output directory to /home/gfbudke/zephyr/mainline/zephyr/twister-out.1
INFO    - Zephyr version: zephyr-v2.6.0-2939-g4ec03e0f2b4a
INFO    - JOBS: 4
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1455 test scenarios (1395 configurations) selected, 1010 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:   51/ 385  13%  skipped: 1030, failed:    0ERROR   - Cmake build failure: /home/gfbudke/zephyr/mainline/zephyr/samples/subsys/usb/dfu for sam4l_ek

ERROR   - sam4l_ek                  samples/subsys/usb/dfu/sample.usb.dfu              FAILED: Cmake build failure
```

```
-- Generated zephyr.dts: /home/gfbudke/zephyr/mainline/zephyr/twister-out/sam4l_ek/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/gfbudke/zephyr/mainline/zephyr/twister-out/sam4l_ek/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/gfbudke/zephyr/mainline/zephyr/twister-out/sam4l_ek/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/gfbudke/zephyr/mainline/zephyr/twister-out/sam4l_ek/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/dts.cmake

warning: FLASH_PAGE_LAYOUT (defined at drivers/flash/Kconfig:54) has direct dependencies FLASH_HAS_PAGE_LAYOUT && FLASH with value n, but is currently being y-selected by the following symbols:
 - STREAM_FLASH (defined at subsys/storage/stream/Kconfig:7), with value y, direct dependencies y (value: y)

warning: FLASH_MAP (defined at subsys/storage/flash_map/Kconfig:10) has direct dependencies FLASH_HAS_DRIVER_ENABLED with value n, but is currently being y-selected by the following symbols:
 - MCUBOOT_IMG_MANAGER (defined at subsys/dfu/Kconfig:25), with value y, direct dependencies <choice> (value: y), and select condition <choice> (value: y)
Parsing /home/gfbudke/zephyr/mainline/zephyr/samples/subsys/usb/dfu/Kconfig
Loaded configuration '/home/gfbudke/zephyr/mainline/zephyr/boards/arm/sam4l_ek/sam4l_ek_defconfig'
Merged configuration '/home/gfbudke/zephyr/mainline/zephyr/samples/subsys/usb/dfu/prj.conf'
```

I mean, there is something on that sample that makes twister select this board.

When I change to below, twister skips dfu test for sam4l_ek, which is wierd.
```
	depends on SOC_FAMILY_SAM || \
		   SOC_SERIES_SAME70 || \
		   SOC_SERIES_SAMV71

```

```
./scripts/twister -p sam4l_ek -s samples/subsys/usb/dfu/sample.usb.dfu
Renaming output directory to /home/gfbudke/zephyr/mainline/zephyr/twister-out.6
INFO    - Zephyr version: zephyr-v2.6.0-2939-g4ec03e0f2b4a
INFO    - JOBS: 4
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1 test scenarios (1 configurations) selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    1/   1  100%  skipped:    1, failed:    0
INFO    - 0 of 0 test configurations passed (0.00%), 0 failed, 1 skipped with 0 warnings in 4.30 seconds
INFO    - In total 0 test cases were executed, 1 skipped on 1 out of total 390 platforms (0.26%)
INFO    - 0 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing xunit report /home/gfbudke/zephyr/mainline/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/gfbudke/zephyr/mainline/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

Reply:
That is by having this change the Kconfig error goes away, but the test I think might still get filtered.  The easiest solution to the sample.usb.dfu is to just exclude sam4l_ek

Reply:
Agreed!